Core/clim-core/table-formatting.lisp

```diff
@@ -54,6 +54,22 @@
 
 (in-package :clim-internals)
 
+;;; Macro argument checking
+
+(eval-when (:compile-toplevel :load-toplevel :execute)
+  (defun maybe-check-constant-keyword-arguments
+      (record-type standard-record-type arguments valid-keywords)
+    (when (and (constantp record-type)
+               (eq (eval record-type) standard-record-type))
+      (loop for (key value) on arguments by #'cddr
+            when (constantp key)
+            do (let ((key (eval key)))
+                 (unless (or (eq key :record-type)
+                             (member key valid-keywords))
+                   (with-current-source-form (arguments)
+                     (error "~@<~S is not a valid initarg for ~S.~@:>"
+                            key standard-record-type))))))))
```

Comment:
since we are doing this, maybe it is worth also verifying, whether the record type has expected protocol class?

Reply:
That seems out of scope for a macro. Somebody might use the macro with a custom record type and define that record type class after using the macro.